test/e2e/test_authcheck.go

```diff
@@ -0,0 +1,126 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	reconcilertesting "github.com/google/knative-gcp/pkg/reconciler/testing"
+	"github.com/google/knative-gcp/test/lib"
+)
+
+const (
+	authCheckServiceAccountName = "auth-check-test"
+	fakeServiceAccount          = "fakeserviceaccount@test-project.iam.gserviceaccount.com"
+	fakeValue                   = "{\n  \"type\": \"service_account\",\n  \"project_id\": \"fake-project-id\",\n  \"private_key_id\": \"fake-key-id\"}"
+)
+
+func AuthCheckForPodCheckTestImpl(t *testing.T, authConfig lib.AuthConfig) {
+	ctx := context.Background()
+	t.Helper()
+	topic, deleteTopic := lib.MakeTopicOrDie(t)
+	defer deleteTopic()
+
+	psName := topic + "-pubsub"
+	svcName := "event-display"
+
+	// Initialize client in workload identity mode, no matter the real mode is workload identity or secret,
+	// so that a default secret will not be created in this test namespace.
+	client := lib.Setup(ctx, t, true, true)
+	defer lib.TearDown(ctx, client)
+
+	serviceAccountName := authConfig.ServiceAccountName
+	var wantMessage string
+	if authConfig.WorkloadIdentity {
+		serviceAccountName = authCheckServiceAccountName
+		so := make([]reconcilertesting.ServiceAccountOption, 0)
+		so = append(so, reconcilertesting.WithServiceAccountAnnotation(fakeServiceAccount))
+		serviceAccount := reconcilertesting.NewServiceAccount(serviceAccountName, client.Namespace, so...)
+		client.CreateServiceAccountOrFail(serviceAccount)
+		wantMessage = "the Pod is not fully authenticated, probably due to corresponding k8s service account and google service account do not establish a correct relationship"
+	} else {
+		so := make([]reconcilertesting.SecretOption, 0)
+		so = append(so, reconcilertesting.WithData(map[string][]byte{
+			"key.json": []byte(fakeValue),
+		}))
+		secret := reconcilertesting.NewSecret("google-cloud-key", client.Namespace, so...)
+		client.CreateSecretOrFail(secret)
+		wantMessage = "error getting the token, probably due to the key stored in the Kubernetes Secret is expired or revoked"
+	}
+
+	pubSubConfig := lib.PubSubConfig{
+		SinkGVK:            lib.ServiceGVK,
+		PubSubName:         psName,
+		SinkName:           svcName,
+		TopicName:          topic,
+		ServiceAccountName: serviceAccountName,
+	}
+
+	lib.MakePubSub(client, pubSubConfig)
+
+	// Authentication check will return error for:
+	// 1. In Workload Identity mode, relationship between ksa and gsa is uncompleted.
+	// 2. In Secret mode, the key in k8s secret is a fake one.
+	gotMessage := client.WaitForSourceAuthCheckPendingOrFail(psName, lib.CloudPubSubSourceV1TypeMeta)
+	if !strings.Contains(gotMessage, wantMessage) {
+		t.Fatalf("unexpected message from PullSubscriptionConditionReady condition with authenticationCheckPending reason: %s", gotMessage)
```

Comment:
Include the want message.

Reply:
I mean include it in the failure message, so that it is easy to see from the test logs why it failed. E.g. 

```suggestion
		t.Fatalf("unexpected message from PullSubscriptionConditionReady condition with authenticationCheckPending reason, want %q, got %q", wantMessage, gotMessage)
```